src/richie/apps/search/filter_definitions/courses.py

```diff
@@ -289,22 +290,54 @@ def get_faceted_definitions(self, facets, data, *args, **kwargs):
         # Get internationalized names for all our keys
         key_i18n_name_map = self.get_i18n_names([*key_count_map])
 
+        # Add human names to keys and counts before sorting as some of our sortings
+        # use alphabetical ordering.
+        values = [
+            # Aggregate the information from right above to build the values
+            {
+                "count": count,
+                "human_name": key_i18n_name_map[key],
+                "key": key,
+            }
+            for key, count in key_count_map.items()
+        ]
+
+        # Sort facets as requested
+        if self.sorting == self.SORTING_NAME:
+            # Alphabetical ascending sorting
+            values = sorted(
+                values,
+                key=lambda value: (value["human_name"], value["count"] * -1),
+            )
+        elif self.sorting == self.SORTING_COUNT:
+            # Sorting by descending facet count
+            values = sorted(
+                values,
+                key=lambda value: (value["count"] * -1, value["human_name"]),
+            )
+        else:
+            # NB: self.SORTING_CONF is not appropriate for Indexables as they are not defined
+            # in the static filter definition but generated from indices.
+            raise ImproperlyConfigured(
+                f'Facet sorting "{self.sorting}" is invalid for filter {self.name}.'
+            )
```

Comment:
Same code a above so could be factorized?

Reply:
It's not exactly the same as in one case we have SORTING_CONF in our cases, whereas in the other it does not exist. Not sure it's worth it to factorize with only 2 slightly differing instances of this code

Reply:
ok ok. I missed the difference.